smartcontractkit/external-adapters-js
# (feat): Add error backoff middleware

## Closes sc-30310

## Description
To prevent repeated failing requests from draining the available API requests per second/minute we can back off from known errors.

## Changes

- Adds a new middleware: `ErrorBackoff`

## Steps to Test

Success case:
1. Run an EA
2. Request any valid request
3. It should return a success
4. Repeat this 3+ times. It should continually return from the cache.

Failure case:
1. Run an EA
2. Request any invalid request
3. It should return an error
4. Repeat this 3 times. On the 3rd, attempt it should return a back off error.
5. Wait one minute and re-request, it should return an error.

## Quality Assurance
